In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

In [3]:
customer = pd.read_excel('Customer.xlsx')
product = pd.read_excel('Product.xlsx')
sales = pd.read_excel('Sales.xlsx')
territories = pd.read_excel('Territories.xlsx')

In [4]:
customer.head()
product.head()
sales.head()
territories.head()

,SalesTerritoryKey,Region,Country,Group,RegionImage,Region Info
0,1,Northwest,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Northwestern_Uni...
1,2,Northeast,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Northeastern_Uni...
2,3,Central,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Central_United_S...
3,4,Southwest,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Southwestern_Uni...
4,5,Southeast,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Southeastern_Uni...


# LIMPEZA DO DATASET

### Substituindo valores nulos

In [5]:
cols_cat = ["SubCategory", "Category", "ModelName"]

product[cols_cat] = product[cols_cat].fillna("Unknown")

In [6]:
product.head()

,ProductKey,ProductName,SubCategory,Category,StandardCost,Color,ListPrice,DaysToManufacture,ProductLine,ModelName,Photo,ProductDescription,StartDate
0,1,Adjustable Race,Unknown,Unknown,NaN,NaN,NaN,0,NaN,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
1,2,Bearing Ball,Unknown,Unknown,NaN,NaN,NaN,0,NaN,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
2,3,BB Ball Bearing,Unknown,Unknown,NaN,NaN,NaN,1,NaN,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
3,4,Headset Ball Bearings,Unknown,Unknown,NaN,NaN,NaN,0,NaN,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
4,5,Blade,Unknown,Unknown,NaN,NaN,NaN,1,NaN,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01


In [7]:
product["Color"] = product["Color"].fillna("Not Specified")

product["ProductLine"] = product["ProductLine"].fillna("Unknown")

### Substituindo valores nulos pela média

In [8]:
product["StandardCost"] = product["StandardCost"].fillna(
    product["StandardCost"].median())

product["ListPrice"] = product["ListPrice"].fillna(
    product["ListPrice"].median())

In [9]:
product.head()

,ProductKey,ProductName,SubCategory,Category,StandardCost,Color,ListPrice,DaysToManufacture,ProductLine,ModelName,Photo,ProductDescription,StartDate
0,1,Adjustable Race,Unknown,Unknown,204.6251,Not Specified,364.09,0,Unknown,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
1,2,Bearing Ball,Unknown,Unknown,204.6251,Not Specified,364.09,0,Unknown,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
2,3,BB Ball Bearing,Unknown,Unknown,204.6251,Not Specified,364.09,1,Unknown,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
3,4,Headset Ball Bearings,Unknown,Unknown,204.6251,Not Specified,364.09,0,Unknown,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01
4,5,Blade,Unknown,Unknown,204.6251,Not Specified,364.09,1,Unknown,Unknown,http://www.avising.com/me/LearnPBI/DataSources...,NaN,1998-06-01


In [10]:
product["ProductDescription"] = product["ProductDescription"].fillna("No description")

In [11]:
product.isna().sum()

ProductKey            0
ProductName           0
SubCategory           0
Category              0
StandardCost          0
Color                 0
ListPrice             0
DaysToManufacture     0
ProductLine           0
ModelName             0
Photo                 0
ProductDescription    0
StartDate             0
dtype: int64

In [12]:
sales.isna().sum()

ProductKey              0
OrderDate               0
ShipDate                0
CustomerKey             0
PromotionKey            0
SalesTerritoryKey       0
SalesOrderNumber        0
SalesOrderLineNumber    0
OrderQuantity           0
UnitPrice               0
TotalProductCost        0
SalesAmount             0
TaxAmt                  0
dtype: int64

In [13]:
territories.isna().sum()

SalesTerritoryKey    0
Region               1
Country              1
Group                1
RegionImage          0
Region Info          1
dtype: int64

In [14]:
territories[territories.isna().any(axis=1)]

,SalesTerritoryKey,Region,Country,Group,RegionImage,Region Info
10,11,NaN,NaN,NaN,http://www.avising.com/me/LearnPBI/DataSources...,NaN


In [15]:
territories['Country']

0      United States
1      United States
2      United States
3      United States
4      United States
5             Canada
6             France
7            Germany
8          Australia
9     United Kingdom
10               NaN
Name: Country, dtype: object

In [16]:
cols = ["Region", "Country", "Group", "Region Info"]

territories[cols] = territories[cols].fillna("Unknown")

Com a falta de informações "Region", "Country", "Group", "Region Info" foi preenchido com "Unknown"

In [17]:
territories.isna().sum()

SalesTerritoryKey    0
Region               0
Country              0
Group                0
RegionImage          0
Region Info          0
dtype: int64

### Unindo os datasets em um só df_sport

In [18]:
df_sport = sales.merge(product, on ="ProductKey", how = "left")

df_sport = df_sport.merge(customer, on = "CustomerKey", how = "left")

df_sport = df_sport.merge(territories, on = "SalesTerritoryKey", how = "left")


In [19]:
df_sport.head()

,ProductKey,OrderDate,ShipDate,CustomerKey,PromotionKey,SalesTerritoryKey,SalesOrderNumber,SalesOrderLineNumber,OrderQuantity,UnitPrice,...,CommuteDistance,CustomerCity,CustomerStateCode,CustomerState,CustomerCountry,Region,Country,Group,RegionImage,Region Info
0,310,2014-01-01,2014-01-08,21768,1,6,SO43697,1,2,1789.1350,...,10+ Miles,Metchosin,BC,British Columbia,Canada,Canada,Canada,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Canada
1,346,2014-01-01,2014-01-08,28389,1,7,SO43698,1,2,1699.9950,...,0-1 Miles,Pantin,93,Seine Saint Denis,France,France,France,Europe,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/France
2,346,2014-01-01,2014-01-08,25863,1,1,SO43699,1,2,1699.9950,...,10+ Miles,Lebanon,OR,Oregon,United States,Northwest,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Northwestern_Uni...
3,336,2014-01-01,2014-01-08,14501,1,4,SO43700,1,2,349.5491,...,1-2 Miles,Beverly Hills,CA,California,United States,Southwest,United States,North America,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Southwestern_Uni...
4,346,2014-01-01,2014-01-08,11003,1,9,SO43701,1,2,1699.9950,...,5-10 Miles,North Ryde,NSW,New South Wales,Australia,Australia,Australia,Pacific,http://www.avising.com/me/LearnPBI/DataSources...,https://en.wikipedia.org/wiki/Australia


### Removendo colunas que não vão ser usadas

In [20]:
df_sport.columns

Index(['ProductKey', 'OrderDate', 'ShipDate', 'CustomerKey', 'PromotionKey',
       'SalesTerritoryKey', 'SalesOrderNumber', 'SalesOrderLineNumber',
       'OrderQuantity', 'UnitPrice', 'TotalProductCost', 'SalesAmount',
       'TaxAmt', 'ProductName', 'SubCategory', 'Category', 'StandardCost',
       'Color', 'ListPrice', 'DaysToManufacture', 'ProductLine', 'ModelName',
       'Photo', 'ProductDescription', 'StartDate', 'FirstName', 'LastName',
       'FullName', 'BirthDate', 'MaritalStatus', 'Gender', 'YearlyIncome',
       'TotalChildren', 'NumberChildrenAtHome', 'Education', 'Occupation',
       'HouseOwnerFlag', 'NumberCarsOwned', 'AddressLine1',
       'DateFirstPurchase', 'CommuteDistance', 'CustomerCity',
       'CustomerStateCode', 'CustomerState', 'CustomerCountry', 'Region',
       'Country', 'Group', 'RegionImage', 'Region Info'],
      dtype='object')

### Contando Produtos 

In [21]:
len(product)

606

In [22]:
df_sport = df_sport.drop(columns=["RegionImage", "Region Info", "AddressLine1","CustomerStateCode","Photo","SalesOrderLineNumber"])

In [23]:
df_sport['NumberCarsOwned'].head() 

0    3
1    0
2    3
3    2
4    1
Name: NumberCarsOwned, dtype: int64

In [24]:
df_sport.head()

,ProductKey,OrderDate,ShipDate,CustomerKey,PromotionKey,SalesTerritoryKey,SalesOrderNumber,OrderQuantity,UnitPrice,TotalProductCost,...,HouseOwnerFlag,NumberCarsOwned,DateFirstPurchase,CommuteDistance,CustomerCity,CustomerState,CustomerCountry,Region,Country,Group
0,310,2014-01-01,2014-01-08,21768,1,6,SO43697,2,1789.1350,2171.2942,...,1,3,2005-07-01,10+ Miles,Metchosin,British Columbia,Canada,Canada,Canada,North America
1,346,2014-01-01,2014-01-08,28389,1,7,SO43698,2,1699.9950,1912.1544,...,0,0,2005-07-01,0-1 Miles,Pantin,Seine Saint Denis,France,France,France,Europe
2,346,2014-01-01,2014-01-08,25863,1,1,SO43699,2,1699.9950,1912.1544,...,1,3,2005-07-01,10+ Miles,Lebanon,Oregon,United States,Northwest,United States,North America
3,336,2014-01-01,2014-01-08,14501,1,4,SO43700,2,349.5491,413.1463,...,1,2,2005-07-01,1-2 Miles,Beverly Hills,California,United States,Southwest,United States,North America
4,346,2014-01-01,2014-01-08,11003,1,9,SO43701,2,1699.9950,1912.1544,...,0,1,2005-07-01,5-10 Miles,North Ryde,New South Wales,Australia,Australia,Australia,Pacific


In [25]:
df_sport.shape

(58189, 44)

### Quantas vendas existem no dataset?

In [26]:
df_sport.describe()

,ProductKey,OrderDate,ShipDate,CustomerKey,PromotionKey,SalesTerritoryKey,OrderQuantity,UnitPrice,TotalProductCost,SalesAmount,TaxAmt,StandardCost,ListPrice,DaysToManufacture,StartDate,YearlyIncome,TotalChildren,NumberChildrenAtHome,HouseOwnerFlag,NumberCarsOwned
count,58189.000000,58189,58189,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000,58189,58189.000000,58189.000000,58189.000000,58189.000000,58189.000000
mean,437.208304,2016-06-03 03:56:09.605939200,2016-06-10 04:03:24.657237760,18853.004640,1.043427,6.261716,1.569386,413.888218,296.539185,503.666270,40.293303,296.539185,503.666270,1.045215,2007-05-14 02:44:51.848974848,59769.887779,1.838921,1.073502,0.690560,1.502466
min,214.000000,2014-01-01 00:00:00,2014-01-08 00:00:00,11000.000000,1.000000,1.000000,1.000000,0.572500,0.856500,2.290000,0.183200,0.856500,2.290000,0.000000,2005-07-01 00:00:00,10000.000000,0.000000,0.000000,0.000000,0.000000
25%,358.000000,2016-04-01 00:00:00,2016-04-08 00:00:00,14012.000000,1.000000,4.000000,1.000000,4.990000,3.362300,8.990000,0.719200,3.362300,8.990000,0.000000,2007-07-01 00:00:00,30000.000000,0.000000,0.000000,0.000000,1.000000
50%,479.000000,2016-07-07 00:00:00,2016-07-14 00:00:00,18151.000000,1.000000,7.000000,1.000000,24.490000,12.192400,32.600000,2.608000,12.192400,32.600000,0.000000,2007-07-01 00:00:00,60000.000000,2.000000,0.000000,1.000000,2.000000
75%,529.000000,2016-10-10 00:00:00,2016-10-17 00:00:00,23450.000000,1.000000,9.000000,2.000000,269.995000,343.649600,539.990000,43.199200,343.649600,539.990000,4.000000,2007-07-01 00:00:00,80000.000000,3.000000,2.000000,1.000000,2.000000
max,606.000000,2016-12-30 00:00:00,2017-01-07 00:00:00,29483.000000,14.000000,10.000000,4.000000,3578.270000,2171.294200,3578.270000,286.261600,2171.294200,3578.270000,4.000000,2007-07-01 00:00:00,170000.000000,5.000000,5.000000,1.000000,4.000000
std,118.099746,NaN,NaN,5433.374315,0.348948,2.960248,1.047532,833.052938,560.171436,941.462817,75.317027,560.171436,941.462817,1.757395,NaN,33128.041818,1.614467,1.580055,0.462267,1.155496


São 58.189 vendas

### Quantos produtos, clientes e regiões diferentes?

In [27]:
df_sport.nunique()

ProductKey                158
OrderDate                1073
ShipDate                 1074
CustomerKey             17918
PromotionKey                4
SalesTerritoryKey          10
SalesOrderNumber        26683
OrderQuantity               3
UnitPrice                 126
TotalProductCost           45
SalesAmount                42
TaxAmt                     42
ProductName               130
SubCategory                17
Category                    3
StandardCost               45
Color                       8
ListPrice                  42
DaysToManufacture           2
ProductLine                 4
ModelName                  40
ProductDescription         40
StartDate                   3
FirstName                 665
LastName                  368
FullName                17838
BirthDate                8160
MaritalStatus               2
Gender                      2
YearlyIncome               16
TotalChildren               6
NumberChildrenAtHome        6
Education                   5
Occupation

São 130 produtos, 17918 clientes e 10 regiões

### Qual o período coberto pelos dados?

In [28]:
df_sport['OrderDate']

0       2014-01-01
1       2014-01-01
2       2014-01-01
3       2014-01-01
4       2014-01-01
           ...    
58184   2016-12-30
58185   2016-12-30
58186   2016-12-30
58187   2016-12-30
58188   2016-12-30
Name: OrderDate, Length: 58189, dtype: datetime64[ns]

De 01/01/2014 à 30/12/2016

### Qual o montante total?

In [29]:
montante = df_sport["SalesAmount"].sum()
print(round(montante,2))

29307836.59


Montante Total de $ 29.307.836.59 dolares

### Vendas por Região

In [30]:
pd.options.display.float_format = '{:,.2f}'.format
sales_by_region = df_sport.groupby("Region")["SalesAmount"].sum()
print(sales_by_region)

Region
Australia        9,051,766.35
Canada           1,966,991.16
Central              3,000.83
France           2,640,525.76
Germany          2,890,707.51
Northeast            6,532.47
Northwest        3,639,701.30
Southeast           12,124.89
Southwest        5,708,995.51
United Kingdom   3,387,490.80
Name: SalesAmount, dtype: float64


In [31]:
df_sport.columns

Index(['ProductKey', 'OrderDate', 'ShipDate', 'CustomerKey', 'PromotionKey',
       'SalesTerritoryKey', 'SalesOrderNumber', 'OrderQuantity', 'UnitPrice',
       'TotalProductCost', 'SalesAmount', 'TaxAmt', 'ProductName',
       'SubCategory', 'Category', 'StandardCost', 'Color', 'ListPrice',
       'DaysToManufacture', 'ProductLine', 'ModelName', 'ProductDescription',
       'StartDate', 'FirstName', 'LastName', 'FullName', 'BirthDate',
       'MaritalStatus', 'Gender', 'YearlyIncome', 'TotalChildren',
       'NumberChildrenAtHome', 'Education', 'Occupation', 'HouseOwnerFlag',
       'NumberCarsOwned', 'DateFirstPurchase', 'CommuteDistance',
       'CustomerCity', 'CustomerState', 'CustomerCountry', 'Region', 'Country',
       'Group'],
      dtype='object')

In [32]:
df_sport['Profit'] = df_sport['SalesAmount'] - df_sport['TotalProductCost']

# Analise Venda & Produto

### Top 10 Produtos por Faturamento

In [33]:
df_produtos = df_sport.groupby(['ProductName', 'SubCategory']).agg(
    Faturamento=('SalesAmount', 'sum'),
    Lucro=('Profit', 'sum'),
    Qtd_Vendida=('OrderQuantity', 'sum')
).reset_index()

df_produtos['Margem_%'] = (df_produtos['Lucro'] / df_produtos['Faturamento'] * 100).round(1)

In [34]:
top_fat = df_produtos.nlargest(10, 'Faturamento')
top_luc = df_produtos.nlargest(15, 'Lucro')

fig = px.bar(top_fat, x='Faturamento', y='ProductName', 
             orientation='h', color='Margem_%',
             color_continuous_scale='RdYlGn',
             title='Top 10 Produtos por Faturamento')
fig.show()

Existe uma relação entre faturamento e margem de lucro, produtos que mais faturaram possuem uma margem maior

### Faturamento e Margem por Categoria

In [56]:
df_margem = df_sport.groupby('Category').agg(
    Faturamento=('SalesAmount', 'sum'),
    Lucro=('Profit', 'sum'),
).reset_index()

df_margem['Margem_%'] = (df_margem['Lucro'] / df_margem['Faturamento'] * 100).round(1)

In [57]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x=df_margem['Category'],
    y=df_margem['Faturamento'],
    name='Faturamento',
    marker_color='steelblue'
))

fig.add_trace(go.Scatter(
    x=df_margem['Category'],
    y=df_margem['Margem_%'],
    name='Margem%',
    yaxis='y2',
    mode='lines+markers',
    line=dict(color='orange',width=2)
))

fig.update_layout(
    title='Faturamento e Margem por Categoria',
    yaxis=dict(title='Faturamento (U$D)'),
    yaxis2=dict(title='Margem % ',overlaying='y', side='right'),
    legend=dict(x=0.01, y=0.99)
)

fig.show()

Bikes lidera em faturamento absoluto, mas sua margem (40,6%) é idêntica à de Clothing, a categoria com menor receita — indicando que escala não gerou vantagem de rentabilidade. Accessories se destaca como exceção: um faturamento absoluto menor, porém a maior margem entre as três categorias, sugerindo que é o segmento mais eficiente proporcionalmente

### Produtos com Maior Volume X Maior Rentabilidade

In [37]:
df_produtos['Rank_Faturamento'] = df_produtos['Faturamento'].rank(ascending=False).astype(int)
df_produtos['Rank_Lucro'] = df_produtos['Lucro'].rank(ascending=False).astype(int)
df_produtos['Diferenca_Rank'] = df_produtos['Rank_Faturamento'] - df_produtos['Rank_Lucro']

In [38]:
fig = px.scatter(
    df_produtos,
    x='Faturamento',
    y='Margem_%',
    size='Qtd_Vendida',
    color='SubCategory',
    hover_name='ProductName',
    title='Volume x Rentabilidade por Produto'
)
fig.add_hline(y=df_produtos['Margem_%'].mean(), line_dash='dash', line_color='gray',
              annotation_text='Margem Média')
fig.add_vline(x=df_produtos['Faturamento'].mean(), line_dash='dash', line_color='gray',
              annotation_text='Faturamento Médio')
fig.show()

Os produtos mais rentáveis não são os mais vendidos — alto volume de vendas não garante alta margem, o que indica que a estratégia de mix de produtos deve priorizar rentabilidade, não apenas giro.

# ANALISE GEOGRAFICA

### Região

In [39]:
df_sport['Country']

0               Canada
1               France
2        United States
3        United States
4            Australia
             ...      
58184        Australia
58185        Australia
58186        Australia
58187        Australia
58188        Australia
Name: Country, Length: 58189, dtype: object

### Criando coluna mês

In [40]:
df_sport['Month'] = df_sport['OrderDate'].dt.month

### Faturamento e Lucro por País

In [41]:
df_pais = df_sport.groupby('Country').agg(
    Faturamento=('SalesAmount', 'sum'),
    Lucro=('Profit', 'sum'),
).reset_index()
df_pais['Margem_%'] = (df_pais['Lucro'] / df_pais['Faturamento'] * 100).round(1)
df_pais = df_pais.sort_values('Faturamento', ascending = False)

In [42]:
fig = px.bar(
    df_pais,
    x = 'Country',
    y = ['Faturamento', 'Lucro'],
    barmode = 'group',
    title = 'Faturamento e Lucro por País',
    text_auto= '.2s'
)
fig.show()

Estados Unidos e Austrália lideram em faturamento, porém apresentam as menores margens de lucro entre os países analisados. Alemanha e França, apesar do menor volume de receita, demonstram melhor eficiência — maior retorno proporcional sobre o faturamento.

### Participação de Faturamento por País

In [43]:
fig = px.pie(
    df_pais,
    names = 'Country',
    values = 'Faturamento',
    title = 'Participação de Faturamento por País',
    hole = 0.6
)

fig.show()

Estados Unidos e Austrália concentram 62,9% do faturamento total, enquanto os quatro países restantes dividem os 37,1% restantes — evidenciando uma distribuição geográfica desigual da receita

### Evolução Anual do Faturamento por País

### Adicionando coluna Ano

In [44]:
df_sport['Year'] = df_sport['OrderDate'].dt.year

In [45]:
df_anual = df_sport.groupby(['Year','Country'])['SalesAmount'].sum().reset_index()

In [46]:
fig = px.line(
    df_anual,
    x = 'Year',
    y = 'SalesAmount',
    color = 'Country',
    markers = True,
    title = 'Evolução Anual do Faturamento por País',
    labels = {'SalesAmount': 'Faturamento (USD)', 'Year': 'Ano'}
)

fig.update_layout(
    xaxis = dict(tickmode = 'array', tickvals = [2014, 2015, 2016])
)

fig.show()

No periodo apresentado nota-se que houve uma queda no faturamento geral de todos os paises no ano de 2015, e que no periodo entre 2015 - 2016 a Australia que liderava com o maior faturamento foi ultrapassada pelo United States, que se tornou o país com maior faturamento em 2016, já o Canada que em 2014 era o quarto maior faturamento, apresentou um movimento negativo, sinalizando uma deterioração consistente no mercado canadense  

# ANALISE TEMPORAL

### Evolução Mensal do Faturamento

In [47]:
df_mensal = df_sport.groupby('Month')['SalesAmount'].sum().reset_index()

fig = px.line(
    df_mensal,
    x = 'Month',
    y = 'SalesAmount',
    markers = True,
    title = 'Evolução Mensal do Faturamento',
    labels = {'SalesAmount': 'Faturamento (USD)', 'Month': 'Período'}
)

fig.update_layout(
    xaxis=dict(
        tickmode = 'array',
        tickvals = df_mensal['Month'][::3].tolist()
    )
)
fig.show()

O faturamento apresenta tendência crescente ao longo do ano, com um pico intermediário em junho seguido de correção em julho — que ainda supera os meses anteriores. A trajetória se mantém ascendente até dezembro, onde atinge o maior valor do período, evidenciando forte sazonalidade no final do ano

### Mês que Mais Faturou 

In [48]:
df_mes = df_sport.groupby(['Year','Month'])['SalesAmount'].sum().reset_index()
df_mes['Month_Nome'] = df_mes['Month'].apply(lambda x: pd.Timestamp(2024, x, 1).strftime('%B'))

melhor_mes = df_mes.loc[df_mes['SalesAmount'].idxmax()]
print(f"Melhor mês: {melhor_mes['Month_Nome']}/{melhor_mes['Year']} - Faturamento: ${melhor_mes['SalesAmount']:,.2f}")

Melhor mês: December/2016 - Faturamento: $1,949,361.11


### Meses que Mais Faturaram por Ano

In [49]:
df_melhor_mes = df_sport.groupby(['Year','Month'])['SalesAmount'].sum().reset_index()
df_melhor_mes['Month_Nome'] = df_melhor_mes['Month'].apply(lambda x: pd.Timestamp(2024, x, 1).strftime('%B'))

resultado = df_melhor_mes.loc[df_melhor_mes.groupby('Year')['SalesAmount'].idxmax()]
print(resultado[['Year','Month_Nome','SalesAmount']].to_string(index = False))

 Year Month_Nome  SalesAmount
 2014       June   755,527.89
 2015       June   577,314.00
 2016   December 1,949,361.11


In [50]:
fig = px.bar(
    df_melhor_mes,
    x = 'Month_Nome',
    y = 'SalesAmount',
    color = 'Year',
    barmode = 'group',
    title = 'Faturamento Mensal por Ano',
    labels = {'SalesAmount': 'Faturamento (USD)','Month_Nome':'Mês','Year': 'Ano'}
)

fig.show()

2016 foi o ano com maior faturamento, superando com folga os anos anteriores em todos os meses. Em contraste, 2015 apresentou queda em relação a 2014 em praticamente todos os meses, configurando um ciclo de retração seguido de forte recuperação em 2016

### Dia Que mais Vendeu

In [51]:
df_dia = df_sport.groupby('OrderDate')['SalesAmount'].sum()

dia_top = df_dia.idxmax()
valor_top = df_dia.max()

print(dia_top, valor_top)

2016-11-30 00:00:00 119822.86


### Top 10 Dias com Maior Volume de Vendas

In [52]:
df_dia = (df_sport.groupby('OrderDate')['SalesAmount'].sum().reset_index().sort_values('SalesAmount',ascending=False).head(10))

df_dia['OrderDate'] = df_dia['OrderDate'].dt.strftime('%d-%m-%Y')
df_dia['SalesAmount_fmt'] = df_dia['SalesAmount'].apply(lambda x: f'${x:^,.0f}')
df_dia = df_dia.sort_values('SalesAmount', ascending = True)

In [53]:
fig = px.bar(
    df_dia,
    x = 'SalesAmount',
    y = 'OrderDate',
    orientation = 'h',
    text = 'SalesAmount_fmt',
    title = 'Top 10 Dias com Maior Volume de Vendas',
    labels = {'SalesAmount': 'Faturamento (USD)', 'OrderDate': 'Data'},
    color='SalesAmount',
    color_continuous_scale = 'Blues'
)

fig.update_traces(textposition = 'outside')

fig.update_layout(
    coloraxis_showscale = False,
    xaxis = dict(tickformat = ',.0f')
)

fig.show()

O ranking revela dois padrões de concentração de vendas: os maiores volumes ocorrem nos últimos dias do mês — influenciados pelo ciclo de recebimento de salários — e oito das dez datas são do último trimestre do ano, refletindo o impacto da sazonalidade de fim de ano no comportamento de compra

### Conclusão Geral

Estados Unidos e Austrália dominam o mercado, concentrando 62,9% do faturamento total. Em 2016 houve recuperação expressiva após retração em 2015, com concentração de vendas no último trimestre evidenciando forte sazonalidade de fim de ano. Bikes lidera em faturamento absoluto, mas Accessories se destaca como o segmento mais eficiente — maior margem proporcional entre as categorias, demonstrando que volume elevado não garante rentabilidade. Esse padrão, porém, não se repete no nível de produto: individualmente, os itens de maior faturamento tendem a apresentar também as maiores margens